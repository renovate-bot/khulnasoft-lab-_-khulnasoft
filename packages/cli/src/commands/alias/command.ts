import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const aliasCommand: Command = {
  name: 'alias',
  description: 'Interact with deployment aliases.',
  arguments: [
    {
      name: 'command',
      required: false,
    },
  ],
  subcommands: [
    {
      name: 'ls',
      description: 'Show all aliases.',
      arguments: [],
      options: [],
      examples: [],
    },
    {
      name: 'set',
      description: 'Create a new alias',
      arguments: [
        {
          name: 'deployment',
          required: true,
        },
        {
          name: 'alias',
          required: true,
        },
      ],
      options: [],
      examples: [],
    },
    {
      name: 'rm',
      description: 'Remove an alias using its hostname.',
      arguments: [
        {
          name: 'alias',
          required: true,
        },
      ],
      options: [],
      examples: [],
    },
  ],
  options: [
    {
      name: 'next',
      description: 'Show next page of results',
      argument: 'MS',
      shorthand: 'n',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'yes',
      description: 'Skip the confirmation prompt when removing an alias',
      shorthand: 'y',
      type: 'boolean',
      deprecated: false,
      multi: false,
    },
    {
      name: 'limit',
      shorthand: 'n',
      description:
        'Number of results to return per page (default: 20, max: 100)',
      argument: 'NUMBER',
      type: 'string',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: 'Add a new alias to `my-api.vercel.app`',
      value: `${packageName} alias set api-ownv3nc9f8.vercel.app my-api.vercel.app`,
    },
    {
      name: 'Custom domains work as alias targets',
      value: `${packageName} alias set api-ownv3nc9f8.vercel.app my-api.com`,
    },
    {
      name: 'The subcommand `set` is the default and can be skipped. Protocols in the URLs are unneeded and ignored',
      value: `${packageName} alias api-ownv3nc9f8.vercel.app my-api.com`,
    },
  ],
};
