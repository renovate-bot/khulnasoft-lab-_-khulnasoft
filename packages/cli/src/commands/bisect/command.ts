import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const bisectCommand: Command = {
  name: 'bisect',
  description: 'Bisect the current project interactively.',
  arguments: [],
  options: [
    {
      name: 'bad',
      description: 'Known bad URL',
      argument: 'URL',
      shorthand: 'b',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'good',
      description: 'Known good URL',
      argument: 'URL',
      shorthand: 'g',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'open',
      description: 'Automatically open each URL in the browser',
      argument: 'URL',
      shorthand: 'o',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'path',
      description: 'Subpath of the deployment URL to test',
      argument: 'URL',
      shorthand: 'p',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'run',
      description: 'Test script to run for each deployment',
      argument: 'URL',
      shorthand: 'r',
      type: 'string',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: 'Bisect the current project interactively',
      value: `${packageName} bisect`,
    },
    {
      name: 'Bisect with a known bad deployment',
      value: `${packageName} bisect --bad example-310pce9i0.vercel.app`,
    },
    {
      name: 'Automated bisect with a run script',
      value: `${packageName} bisect --run ./test.sh`,
    },
  ],
};
