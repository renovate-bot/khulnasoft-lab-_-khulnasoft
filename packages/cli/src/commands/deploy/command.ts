import { Command } from '../help';

export const deployCommand: Command = {
  name: 'deploy',
  description:
    'Deploy your project to Vercel. The `deploy` command is the default command for the Vercel CLI, and can be omitted (`vc deploy my-app` equals `vc my-app`).',
  arguments: [
    {
      name: 'project-path',
      required: false,
    },
  ],
  options: [
    {
      name: 'force',
      shorthand: 'f',
      type: 'boolean',
      deprecated: false,
      description: 'Force a new deployment even if nothing has changed',
      multi: false,
    },
    {
      name: 'with-cache',
      shorthand: null,
      type: 'boolean',
      deprecated: false,
      description: 'Retain build cache when using "--force"',
      multi: false,
    },
    {
      name: 'public',
      shorthand: 'p',
      type: 'boolean',
      deprecated: false,
      description: 'Deployment is public (`/_src`) is exposed)',
      multi: false,
    },
    {
      name: 'env',
      shorthand: 'e',
      type: 'string',
      argument: 'key=value',
      deprecated: false,
      multi: true,
      description:
        'Specify environment variables during run-time (e.g. `-e KEY1=value1 -e KEY2=value2`)',
    },
    {
      name: 'build-env',
      shorthand: 'b',
      type: 'string',
      argument: 'key=value',
      deprecated: false,
      multi: true,
      description:
        'Specify environment variables during build-time (e.g. `-b KEY1=value1 -b KEY2=value2`)',
    },
    {
      name: 'meta',
      shorthand: 'm',
      type: 'string',
      argument: 'key=value',
      deprecated: false,
      multi: true,
      description:
        'Specify metadata for the deployment (e.g. `-m KEY1=value1 -m KEY2=value2`)',
    },
    {
      name: 'regions',
      shorthand: null,
      type: 'string',
      deprecated: false,
      description: 'Set default regions to enable the deployment on',
      multi: false,
    },
    {
      name: 'prebuilt',
      shorthand: null,
      type: 'boolean',
      deprecated: false,
      description:
        'Use in combination with `vc build`. Deploy an existing build',
      multi: false,
    },
    {
      name: 'prod',
      shorthand: null,
      type: 'boolean',
      deprecated: false,
      description: 'Create a production deployment',
      multi: false,
    },
    {
      name: 'archive',
      shorthand: null,
      type: 'string',
      deprecated: false,
      description:
        'Compress the deployment code into a file before uploading it',
      multi: false,
    },
    {
      name: 'no-wait',
      shorthand: null,
      type: 'boolean',
      deprecated: false,
      description: "Don't wait for the deployment to finish",
      multi: false,
    },
    {
      name: 'skip-domain',
      shorthand: null,
      type: 'boolean',
      deprecated: false,
      description: undefined,
      multi: false,
    },
    {
      name: 'yes',
      shorthand: 'y',
      type: 'boolean',
      deprecated: false,
      description: 'Use default options to skip all prompts',
      multi: false,
    },
    {
      name: 'name',
      shorthand: 'n',
      type: 'string',
      deprecated: true,
      description: 'Provide a Vercel Project name',
      multi: false,
    },
    {
      name: 'no-clipboard',
      shorthand: null,
      type: 'boolean',
      deprecated: true,
      description: 'Do not copy deployment URL to clipboard',
      multi: false,
    },
    {
      name: 'target',
      shorthand: null,
      type: 'string',
      deprecated: true,
      description: 'Specify the target deployment environment',
      multi: false,
    },
    {
      name: 'confirm',
      shorthand: 'c',
      type: 'boolean',
      deprecated: true,
      description: 'Use default options to skip all prompts',
      multi: false,
    },
  ],
  examples: [
    {
      name: 'Deploy the current directory',
      value: 'vercel',
    },
    {
      name: 'Deploy a custom path',
      value: 'vercel /usr/src/project',
    },
    {
      name: 'Deploy with run-time Environment Variables',
      value: 'vercel -e NODE_ENV=production',
    },
    {
      name: 'Deploy with prebuilt outputs',
      value: ['vercel build', 'vercel deploy --prebuilt'],
    },
    {
      name: 'Write Deployment URL to a file',
      value: 'vercel > deployment-url.txt',
    },
  ],
};
