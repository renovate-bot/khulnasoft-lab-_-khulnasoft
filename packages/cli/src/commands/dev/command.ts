import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const devCommand: Command = {
  name: 'dev',
  description: `Starts the \`${packageName} dev\` server.`,
  arguments: [
    {
      name: 'dir',
      required: false,
    },
  ],
  options: [
    {
      name: 'listen',
      description: 'Specify a URI endpoint on which to listen [0.0.0.0:3000]',
      argument: 'uri',
      shorthand: null,
      type: 'string',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: `Start the \`${packageName} dev\` server on port 8080`,
      value: `${packageName} dev --listen 8080`,
    },
    {
      name: 'Make the `vercel dev` server bind to localhost on port 5000',
      value: `${packageName} dev --listen 127.0.0.1:5000 `,
    },
  ],
};
