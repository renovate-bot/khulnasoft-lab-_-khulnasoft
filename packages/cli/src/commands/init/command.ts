import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const initCommand: Command = {
  name: 'init',
  description: 'Initialize example Vercel Projects',
  arguments: [
    {
      name: 'example',
      required: false,
    },
    {
      name: 'dir',
      required: false,
    },
  ],
  subcommands: undefined,
  options: [
    {
      name: 'force',
      description: 'Overwrite destination directory if exists [off]',
      argument: undefined,
      shorthand: 'f',
      type: 'boolean',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: 'Choose from all available examples',
      value: `${packageName} init`,
    },
    {
      name: 'Initialize example project into a new directory',
      value: `${packageName} init <example>`,
    },
    {
      name: 'Initialize example project into specified directory',
      value: `${packageName} <example> <dir>`,
    },
    {
      name: 'Initialize example project without checking',
      value: `${packageName} init <example> --force`,
    },
  ],
};
