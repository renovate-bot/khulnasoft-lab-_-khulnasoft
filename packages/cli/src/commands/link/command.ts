import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const linkCommand: Command = {
  name: 'link',
  description: 'Link a local directory to a Vercel Project.',
  arguments: [],
  options: [
    {
      name: 'repo',
      description: 'Link multiple projects based on Git repository (alpha)',
      shorthand: 'r',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'project',
      description: 'Specify a project name',
      shorthand: 'p',
      argument: 'NAME',
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'yes',
      description:
        'Skip questions when setting up new project using default scope and settings',
      shorthand: 'y',
      type: 'boolean',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: 'Link current directory to a Vercel Project',
      value: `${packageName} link`,
    },
    {
      name: 'Link current directory with default options and skip questions',
      value: `${packageName} link --yes`,
    },
    {
      name: 'Link a specific directory to a Vercel Project',
      value: `${packageName} link --cwd /path/to/project`,
    },
    {
      name: 'Link to the current Git repository, allowing for multiple Vercel Projects to be linked simultaneously (useful for monorepos)',
      value: `${packageName} link --repo`,
    },
  ],
};
