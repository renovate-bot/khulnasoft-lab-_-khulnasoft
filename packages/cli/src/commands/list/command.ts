import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const listCommand: Command = {
  name: 'list',
  description: 'List app deployments for an app.',
  arguments: [
    {
      name: 'app',
      required: false,
    },
  ],
  options: [
    {
      name: 'meta',
      description:
        'Filter deployments by metadata (e.g.: `-m KEY=value`). Can appear many times.',
      argument: 'KEY=value',
      shorthand: null,
      type: 'string',
      deprecated: false,
      multi: true,
    },
    {
      name: 'environment',
      description: '',
      argument: 'production|preview',
      shorthand: null,
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'next',
      description: 'Show next page of results',
      argument: 'MS',
      shorthand: 'n',
      type: 'string',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: 'List all deployments for the currently linked project',
      value: `${packageName} list`,
    },
    {
      name: 'List all deployments for the project `my-app` in the team of the currently linked project',
      value: `${packageName} list my-app`,
    },
    {
      name: 'Filter deployments by metadata',
      value: `${packageName} list -m key1=value1 -m key2=value2`,
    },
    {
      name: 'Paginate deployments for a project, where `1584722256178` is the time in milliseconds since the UNIX epoch',
      value: `${packageName} list my-app --next 1584722256178`,
    },
  ],
};
