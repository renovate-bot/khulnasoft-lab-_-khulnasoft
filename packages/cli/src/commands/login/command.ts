import { Command } from '../help';
import { packageName } from '../../util/pkg-name';

export const loginCommand: Command = {
  name: 'login',
  description: 'Authenticate using your email or team id.',
  arguments: [
    {
      name: 'email or team id',
      required: false,
    },
  ],
  options: [
    {
      name: 'github',
      description: 'Log in with GitHub',
      shorthand: null,
      type: 'string',
      deprecated: false,
      multi: false,
    },
    {
      name: 'oob',
      description: 'Log in with "out of band" authentication',
      shorthand: null,
      type: 'string',
      deprecated: false,
      multi: false,
    },
  ],
  examples: [
    {
      name: 'Log into the Vercel platform',
      value: `${packageName} login`,
    },
    {
      name: 'Log in using a specific email address',
      value: `${packageName} login username@example.com`,
    },
    {
      name: 'Log in using a specific team "slug" for SAML Single Sign-On',
      value: `${packageName} login acme`,
    },
    {
      name: 'Log in using GitHub in "out-of-band" mode',
      value: `${packageName} login --github --oob`,
    },
  ],
};
