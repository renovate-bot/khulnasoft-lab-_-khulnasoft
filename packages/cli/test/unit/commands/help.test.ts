import {
  help,
  lineToString,
  outputArrayToString,
} from '../../../src/commands/help';
import { deployCommand } from '../../../src/commands/deploy/command';

describe('help command', () => {
  describe('lineToString', () => {
    test.each([
      {
        line: ['a', 'b', 'c'],
        expected: 'a b c',
      },
      {
        line: [' ', 'a', ' ', 'b', ' ', 'c', ' '],
        expected: ' a b c ',
      },
      {
        line: [' ', '  ', '   '],
        expected: '      ',
      },
      {
        line: ['a', '  ', '   ', 'b', 'c'],
        expected: 'a     b c',
      },
    ])(
      'should insert spaces between non-whitespace items only; $line',
      ({ line, expected }) => {
        expect(lineToString(line)).toBe(expected);
      }
    );
  });

  describe('outputArrayToString', () => {
    test('should join a list of strings using newlines', () => {
      expect(outputArrayToString(['line 1', 'line 2', 'line 3'])).toBe(
        'line 1\nline 2\nline 3'
      );
    });
  });

  describe('help output snapshots', () => {
    test.each([40, 80, 120])('column width %i', width => {
      expect(help(deployCommand, { columns: width })).toMatchSnapshot();
    });
  });
});
