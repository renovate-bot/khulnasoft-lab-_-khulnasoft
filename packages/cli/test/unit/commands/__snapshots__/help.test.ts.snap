// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help command help output snapshots column width 40 1`] = `
"
  â–² [1mvercel[22m [1mdeploy[22m [project-path] [options]

  Deploy your project to Vercel. The    
  \`deploy\` command is the default       
  command for the Vercel CLI, and can   
  be omitted (\`vc deploy my-app\` equals 
  \`vc my-app\`).                         

  [2mOptions[22m:

       --archive                Compress
                                the     
                                deploymâ€¦
                                code    
                                into a  
                                file    
                                before  
                                uploadiâ€¦
                                it      
  -b,  --build-env <key=value>  Specify 
                                environâ€¦
                                variablâ€¦
                                during  
                                build-tâ€¦
                                (e.g.   
                                \`-b     
                                KEY1=vaâ€¦
                                -b      
                                KEY2=vaâ€¦
  -e,  --env <key=value>        Specify 
                                environâ€¦
                                variablâ€¦
                                during  
                                run-time
                                (e.g.   
                                \`-e     
                                KEY1=vaâ€¦
                                -e      
                                KEY2=vaâ€¦
  -f,  --force                  Force a 
                                new     
                                deploymâ€¦
                                even if 
                                nothing 
                                has     
                                changed 
  -m,  --meta <key=value>       Specify 
                                metadata
                                for the 
                                deploymâ€¦
                                (e.g.   
                                \`-m     
                                KEY1=vaâ€¦
                                -m      
                                KEY2=vaâ€¦
       --no-wait                Don't   
                                wait for
                                the     
                                deploymâ€¦
                                to      
                                finish  
       --prebuilt               Use in  
                                combinaâ€¦
                                with \`vc
                                build\`. 
                                Deploy  
                                an      
                                existing
                                build   
       --prod                   Create a
                                productâ€¦
                                deploymâ€¦
  -p,  --public                 Deploymâ€¦
                                is      
                                public  
                                (\`/_srcâ€¦
                                is      
                                exposed)
       --regions                Set     
                                default 
                                regions 
                                to      
                                enable  
                                the     
                                deploymâ€¦
                                on      
       --with-cache             Retain  
                                build   
                                cache   
                                when    
                                using   
                                "--forcâ€¦
  -y,  --yes                    Use     
                                default 
                                options 
                                to skip 
                                all     
                                prompts 


  [2mGlobal Options[22m:

       --cwd <DIR>            Sets the  
                              current   
                              working   
                              directory 
                              for a     
                              single run
                              of a      
                              command   
  -d,  --debug                Debug mode
                              (default  
                              off)      
  -Q,  --global-config <DIR>  Path to   
                              the global
                              \`.vercel\` 
                              directory 
  -h,  --help                 Output    
                              usage     
                              informatiâ€¦
  -A,  --local-config <FILE>  Path to   
                              the local 
                              \`vercel.jâ€¦
                              file      
       --no-color             No color  
                              mode      
                              (default  
                              off)      
  -S,  --scope                Set a     
                              custom    
                              scope     
  -t,  --token <TOKEN>        Login     
                              token     
  -v,  --version              Output the
                              version   
                              number    


  [2mExamples:[22m

  [90m-[39m Deploy the current directory

    [36m$ vercel[39m

  [90m-[39m Deploy a custom path

    [36m$ vercel /usr/src/project[39m

  [90m-[39m Deploy with run-time Environment Variables

    [36m$ vercel -e NODE_ENV=production[39m

  [90m-[39m Deploy with prebuilt outputs

    [36m$ vercel build[39m
    [36m$ vercel deploy --prebuilt[39m

  [90m-[39m Write Deployment URL to a file

    [36m$ vercel > deployment-url.txt[39m

"
`;

exports[`help command help output snapshots column width 80 1`] = `
"
  â–² [1mvercel[22m [1mdeploy[22m [project-path] [options]

  Deploy your project to Vercel. The \`deploy\` command is the default command    
  for the Vercel CLI, and can be omitted (\`vc deploy my-app\` equals \`vc         
  my-app\`).                                                                     

  [2mOptions[22m:

       --archive                Compress the deployment code into a file before 
                                uploading it                                    
  -b,  --build-env <key=value>  Specify environment variables during build-time 
                                (e.g. \`-b KEY1=value1 -b KEY2=value2\`)          
  -e,  --env <key=value>        Specify environment variables during run-time   
                                (e.g. \`-e KEY1=value1 -e KEY2=value2\`)          
  -f,  --force                  Force a new deployment even if nothing has      
                                changed                                         
  -m,  --meta <key=value>       Specify metadata for the deployment (e.g. \`-m   
                                KEY1=value1 -m KEY2=value2\`)                    
       --no-wait                Don't wait for the deployment to finish         
       --prebuilt               Use in combination with \`vc build\`. Deploy an   
                                existing build                                  
       --prod                   Create a production deployment                  
  -p,  --public                 Deployment is public (\`/_src\`) is exposed)      
       --regions                Set default regions to enable the deployment on 
       --with-cache             Retain build cache when using "--force"         
  -y,  --yes                    Use default options to skip all prompts         


  [2mGlobal Options[22m:

       --cwd <DIR>            Sets the current working directory for a single   
                              run of a command                                  
  -d,  --debug                Debug mode (default off)                          
  -Q,  --global-config <DIR>  Path to the global \`.vercel\` directory            
  -h,  --help                 Output usage information                          
  -A,  --local-config <FILE>  Path to the local \`vercel.json\` file              
       --no-color             No color mode (default off)                       
  -S,  --scope                Set a custom scope                                
  -t,  --token <TOKEN>        Login token                                       
  -v,  --version              Output the version number                         


  [2mExamples:[22m

  [90m-[39m Deploy the current directory

    [36m$ vercel[39m

  [90m-[39m Deploy a custom path

    [36m$ vercel /usr/src/project[39m

  [90m-[39m Deploy with run-time Environment Variables

    [36m$ vercel -e NODE_ENV=production[39m

  [90m-[39m Deploy with prebuilt outputs

    [36m$ vercel build[39m
    [36m$ vercel deploy --prebuilt[39m

  [90m-[39m Write Deployment URL to a file

    [36m$ vercel > deployment-url.txt[39m

"
`;

exports[`help command help output snapshots column width 120 1`] = `
"
  â–² [1mvercel[22m [1mdeploy[22m [project-path] [options]

  Deploy your project to Vercel. The \`deploy\` command is the default command for the Vercel CLI, and can be omitted     
  (\`vc deploy my-app\` equals \`vc my-app\`).                                                                              

  [2mOptions[22m:

       --archive                Compress the deployment code into a file before uploading it                            
  -b,  --build-env <key=value>  Specify environment variables during build-time (e.g. \`-b KEY1=value1 -b KEY2=value2\`)  
  -e,  --env <key=value>        Specify environment variables during run-time (e.g. \`-e KEY1=value1 -e KEY2=value2\`)    
  -f,  --force                  Force a new deployment even if nothing has changed                                      
  -m,  --meta <key=value>       Specify metadata for the deployment (e.g. \`-m KEY1=value1 -m KEY2=value2\`)              
       --no-wait                Don't wait for the deployment to finish                                                 
       --prebuilt               Use in combination with \`vc build\`. Deploy an existing build                            
       --prod                   Create a production deployment                                                          
  -p,  --public                 Deployment is public (\`/_src\`) is exposed)                                              
       --regions                Set default regions to enable the deployment on                                         
       --with-cache             Retain build cache when using "--force"                                                 
  -y,  --yes                    Use default options to skip all prompts                                                 


  [2mGlobal Options[22m:

       --cwd <DIR>            Sets the current working directory for a single run of a command                          
  -d,  --debug                Debug mode (default off)                                                                  
  -Q,  --global-config <DIR>  Path to the global \`.vercel\` directory                                                    
  -h,  --help                 Output usage information                                                                  
  -A,  --local-config <FILE>  Path to the local \`vercel.json\` file                                                      
       --no-color             No color mode (default off)                                                               
  -S,  --scope                Set a custom scope                                                                        
  -t,  --token <TOKEN>        Login token                                                                               
  -v,  --version              Output the version number                                                                 


  [2mExamples:[22m

  [90m-[39m Deploy the current directory

    [36m$ vercel[39m

  [90m-[39m Deploy a custom path

    [36m$ vercel /usr/src/project[39m

  [90m-[39m Deploy with run-time Environment Variables

    [36m$ vercel -e NODE_ENV=production[39m

  [90m-[39m Deploy with prebuilt outputs

    [36m$ vercel build[39m
    [36m$ vercel deploy --prebuilt[39m

  [90m-[39m Write Deployment URL to a file

    [36m$ vercel > deployment-url.txt[39m

"
`;
